src/Mvc/Mvc.ViewFeatures/src/Rendering/SystemTextJsonHelper.cs

```diff
@@ -0,0 +1,28 @@
+﻿
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Html;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    internal class SystemTextJsonHelper : IJsonHelper
+    {
+        private readonly MvcOptions _mvcOptions;
+
+        public SystemTextJsonHelper(IOptions<MvcOptions> mvcOptions)
+        {
+            _mvcOptions = mvcOptions.Value;
+        }
+
+        /// <inheritdoc />
+        public IHtmlContent Serialize(object value)
+        {
+            // JsonSerializer always encodes
+            var json = JsonSerializer.ToString(value, value?.GetType() ?? typeof(object), _mvcOptions.SerializerOptions);
```

Comment:
Hmmm, is there no overload on JsonSerializer that just takes a value?

Reply:
I was under the impression that https://github.com/dotnet/corefx/issues/36166 wasn't done. But it works for `object value`. Updated